File: src/musikdroid/app/src/main/java/io/casey/musikcube/remote/websocket/Prefs.java
package io.casey.musikcube.remote.websocket;

public interface Prefs {
    String NAME = "prefs";

    interface Key {
        String ADDRESS = "address";
        String MAIN_PORT = "port";
        String AUDIO_PORT = "http_port";
        String PASSWORD = "password";
        String ALBUM_ART_ENABLED = "album_art_enabled";
        String MESSAGE_COMPRESSION_ENABLED = "message_compression_enabled";
        String STREAMING_PLAYBACK = "streaming_playback";
        String SOFTWARE_VOLUME = "software_volume";
        String SSL_ENABLED = "ssl_enabled";
        String CERT_VALIDATION_DISABLED = "cert_validation_disabled";
        String TRANSCODER_BITRATE_INDEX = "transcoder_bitrate_index";
        String DISK_CACHE_SIZE_INDEX = "disk_cache_size_index";
    }

    interface Default {
        String ADDRESS = "192.168.1.100";
        int MAIN_PORT = 7905;
        int AUDIO_PORT = 7906;
        String PASSWORD = "";
        boolean ALBUM_ART_ENABLED = true;
        boolean MESSAGE_COMPRESSION_ENABLED = true;
        boolean STREAMING_PLAYBACK = false;
        boolean SOFTWARE_VOLUME = false;
        boolean SSL_ENABLED = false;
        boolean CERT_VALIDATION_DISABLED = false;
        int TRANSCODER_BITRATE_INDEX = 0;
        int DISK_CACHE_SIZE_INDEX = 0;
    }
}

File: src/musikdroid/app/src/main/java/io/casey/musikcube/remote/websocket/Prefs.kt
package io.casey.musikcube.remote.websocket

class Prefs {
    interface Key {
        companion object {
            val ADDRESS = "address"
            val MAIN_PORT = "port"
            val AUDIO_PORT = "http_port"
            val PASSWORD = "password"
            val ALBUM_ART_ENABLED = "album_art_enabled"
            val MESSAGE_COMPRESSION_ENABLED = "message_compression_enabled"
            val STREAMING_PLAYBACK = "streaming_playback"
            val SOFTWARE_VOLUME = "software_volume"
            val SSL_ENABLED = "ssl_enabled"
            val CERT_VALIDATION_DISABLED = "cert_validation_disabled"
            val TRANSCODER_BITRATE_INDEX = "transcoder_bitrate_index"
            val DISK_CACHE_SIZE_INDEX = "disk_cache_size_index"
            val SYSTEM_SERVICE_FOR_REMOTE = "system_service_for_remote"
            val UPDATE_DIALOG_SUPPRESSED_VERSION = "update_dialog_suppressed_version"
        }
    }

    interface Default {
        companion object {
            val ADDRESS = "192.168.1.100"
            val MAIN_PORT = 7905
            val AUDIO_PORT = 7906
            val PASSWORD = ""
            val ALBUM_ART_ENABLED = true
            val MESSAGE_COMPRESSION_ENABLED = true
            val STREAMING_PLAYBACK = false
            val SOFTWARE_VOLUME = false
            val SSL_ENABLED = false
            val CERT_VALIDATION_DISABLED = false
            val TRANSCODER_BITRATE_INDEX = 0
            val DISK_CACHE_SIZE_INDEX = 0
            val SYSTEM_SERVICE_FOR_REMOTE = false
            val UPDATE_DIALOG_SUPPRESSED_VERSION = ""
        }
    }

    companion object {
        val NAME = "prefs"
    }
}
